synapse/lib/stormtypes.py

```diff
@@ -7177,6 +7194,20 @@ async def liftByProp(self, propname, propvalu=None, propcmpr='='):
         async for _, buid, sode in layr.liftByPropValu(liftform, liftprop, cmprvals):
             yield await self.runt.snap._joinStorNode(buid, {iden: sode})
 
+    @stormfunc(readonly=True)
+    async def liftByData(self, name):
+
+        name = await tostr(name)
+
+        iden = self.valu.get('iden')
+        layr = self.runt.snap.core.getLayer(iden)
+
+        await self.runt.reqUserCanReadLayer(iden)
+
+        async for _, buid, sode in layr.liftByDataName(name):
+            yield await self.runt.snap._joinStorNode(buid, {iden: sode})
+            return
```

Comment:
stray return?

Reply:
Tells me we'll need a multi-node lift test.